llama_hub/tools/metaphor/base.py

```diff
@@ -0,0 +1,100 @@
+"""Metaphor tool spec."""
+
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from typing import Optional, List
+from llama_index.readers.schema.base import Document
+import requests
+import urllib.parse
+import datetime
+
+class MetaphorToolSpec(BaseToolSpec):
+    """Metaphor tool spec."""
+
+    spec_functions = ["metaphor_search", "retrieve_documents", "current_date", "find_similar"]
+
+    def __init__(self, api_key: str) -> None:
+        """Initialize with parameters."""
+        from metaphor_python import Metaphor
+        self.client = Metaphor(api_key=api_key)
+
+    def metaphor_search(
+        self,
+        query: str,
+        num_results: Optional[int] = 3,
+        include_domains: Optional[List[str]] = None,
+        exclude_domains: Optional[List[str]] = None,
+        start_published_date: Optional[str] = None,
+        end_published_date: Optional[str] = None
+    ) -> str:
+        """
+        Metaphor allows you to use a natural language query to search the internet
+
+        Args:
+            query (str): A natural language query phrased as an answer for what the link provides, ie: "This is the latest news about space"
```

Comment:
This will need heavy prompt engineering in order for something like GPT3.5 to form proper queries. We have the proper prompts and could give them. Alternatively, the "autoprompt" flag can be used, and we use our own LLMs to form the query. Thoughts on what the better pattern would be? I'm a little worried about too lengthy of a prompt crowding the context window here - it would be about 500 tokens. Not sure what kind of context window budget considerations are at play.

Reply:
I think it makes sense to enable autoprompt, no point in trying to prompt the LLM twice; will update 👍